src/shared/Validation.js

```diff
@@ -25,9 +25,80 @@ export const validateOnInput = (
   const formatErrors = sectionFormatErrors.slice();
 
   switch (fieldType) {
-    case 'number': {
+    case 'int16': {
       const idx = formatErrors.indexOf(FORM_ERRORS.INVALID_FORMAT);
-      if (input && !validator.isInt(input)) {
+      const validInt = validator.isInt(input);
```

Comment:
`validator` only validates strings, and will throw an error otherwise, which could potentially cause a lot of problems. Either surround this with a `try-catch`, or ensure that `input` will be a string.

https://github.com/chriso/validator.js/blob/master/src/lib/util/assertString.js

Further, `isInt` takes a second optional parameter where you can specify a min and a max:

https://github.com/chriso/validator.js/blob/master/src/lib/isInt.js

I'd recommend removing the check below, and replace it with a single `validator` check:

```
const isValid = validator.isInt(input, {
  max: INT_SHORT_SIGNED_MAX_VALUE,
  min: INT_SHORT_SIGNED_MIN_VALUE
})
```

Reply:
Fixed